integration-tests/benches/sign.rs

```diff
@@ -0,0 +1,239 @@
+use criterion::measurement::ValueFormatter;
+use criterion::Criterion;
+use criterion::{measurement::Measurement, Throughput};
+use integration_tests::cluster::{self, Cluster};
+use std::sync::Arc;
+use std::sync::Mutex;
+use std::time::Duration;
+use std::time::Instant;
+
+pub const SIGNATURE_AMOUNT: usize = 30;
+
+struct NodeTimeMeasurement {
+    name: &'static str,
+    data: Vec<f64>,
+    at: Mutex<usize>,
+}
+
+impl Measurement for NodeTimeMeasurement {
+    type Intermediate = f64;
+    type Value = Duration;
+
+    fn start(&self) -> Self::Intermediate {
+        let mut at = self.at.lock().unwrap();
+        let value = self.data[*at];
+        *at += 1;
+        if *at >= self.data.len() {
+            *at = 0;
+        }
+        value
+    }
+
+    fn end(&self, i: Self::Intermediate) -> Self::Value {
+        // Here, instead of using `elapsed`, you can fetch external runtime data points.
+        // For example, you could read from a Prometheus metric, log file, or database.
+        Duration::from_secs_f64(i)
+    }
+
+    fn add(&self, a: &Self::Value, b: &Self::Value) -> Self::Value {
+        *a + *b
+    }
+
+    fn zero(&self) -> Self::Value {
+        Duration::new(0, 0)
+    }
+
+    fn to_f64(&self, value: &Self::Value) -> f64 {
+        value.as_nanos() as f64
+    }
+
+    fn formatter(&self) -> &dyn criterion::measurement::ValueFormatter {
+        &DurationFormatter
+    }
+}
+
+fn bench_on_metrics(measurement: NodeTimeMeasurement) {
+    let name = measurement.name.to_string();
+    let mut c = Criterion::default()
+        .sample_size(measurement.data.len())
+        .warm_up_time(Duration::from_nanos(1))
+        .measurement_time(Duration::from_millis(1))
+        .with_measurement(measurement);
+
+    let mut group = c.benchmark_group("bench_on_metrics");
+    group.bench_function(name, |b| {
+        b.iter(|| {
+            std::thread::sleep(Duration::from_millis(1)); // Simulating work
+        })
+    });
+    group.finish();
+}
+
+fn main() {
+    let rt = tokio::runtime::Builder::new_multi_thread()
+        .enable_all()
+        .build()
+        .unwrap();
+    let started = Instant::now();
+    let (nodes, accounts) = rt.block_on(async {
+        let nodes = cluster::spawn()
+            .with_config(|cfg| {
+                cfg.protocol.triple.min_triples = SIGNATURE_AMOUNT as u32 * 4;
```

Comment:
How do we control triple pre-generation?

Reply:
just setting the stockpile amount or calling `cluster::spawn().prestockpile(multiplier)` where `multiplier` creates a multiple of the triple stockpile